client/core/trade.go

```diff
@@ -2853,6 +2856,11 @@ func (t *trackedTrade) redeemFee() uint64 {
 // This method accesses match fields and MUST be called with the trackedTrade
 // mutex lock held for writes.
 func (t *trackedTrade) confirmRedemption(match *matchTracker) {
+	if len(match.MetaData.Proof.Auth.RedeemSig) == 0 { // shouldConfirmRedemption should prevent this
+		t.dc.log.Warnf("Not confirming match %v before it is reported to server", match)
+		return // let resendPendingRequests have it
+	}
+
```

Comment:
This prevents `ConfirmRedemption` from happening until the server can be contacted? Maybe the wallet needs to send a new transaction and that is time sensitive. Is it possible to allow that to continue anyway?

Also in the unlikely event the server disappears, would this trade stay unconfirmed forever?

Reply:
Good point.  Will have to approach this differently, probably using the self governed state we added previously to identify when to confirm regardless

Reply:
Even if the trade is not self governed, the server may not be able to find the redemption due to an issue for which the redemption needs to be resubmitted. I don't think the redemption confirmation check should ever have to wait on the server.

Reply:
On a side note, our `ConfirmRedemption` implementation for BTC is very unlikely to work if it gets to "resubmitting" the transaction.  In most case, you probably just need to rebroadcast the original. Submitting a new one (different txid) is very likely to be rejected by the network as a double spend.  Even if it is not rejected, the original one could still be mined (or maybe already was and we just didn't see it for whatever reasons) and we'd never be able to confirm anything.

The pathological case I was addressing with this PR was actually needing to just rebroadcast the original (or wait for it to propagate / rebroadcast automatically in time).  The redeem request to the server was failing because server couldn't see it.  Ultimately it was mined on a different best block that the server had, and it took a reorg for it to land on the same chain the server was following.  But since the match was considered confirmed, the client never tried to send the request again.
It's worse on 0.6 because BTC and DCR confirm automatically (no checks).